compiler/src/iree/compiler/Codegen/Dialect/Codegen/IR/IREECodegenEnums.h

```diff
@@ -0,0 +1,19 @@
+// Copyright 2025 The IREE Authors
```

Comment:
Dead file? I don't see any codegen enums anymore

Reply:
This is harmonization so that we don't have all this in LoweringConfigEnums.h.inc which just gets included in the attrs file

Reply:
I did not get a chance to revisit Codegen dialect, but I like how we define these in IREE core dialects better, e.g., move all attributes/interfaces/enums/etc to `IREECodegenTypes.h`. I think we don't really need a separation between attrs.h, interfaces.h, and enums.h.

E.g.,

https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Dialect/Encoding/IR/EncodingTypes.h

https://github.com/iree-org/iree/blob/main/compiler/src/iree/compiler/Codegen/Dialect/CPU/IR/IREECPUTypes.h

Reply:
> GPU has An Enums.h, which was what I was going off of
> And a lot of upstream dialects have a separate Enums.h

If it were me, I'd remove the `Enums.h` from GPU. Because I don't see a separation need in most of the dialects.

I'm not going to say if we should follow upstream or not, because I don't see a need for the separation. However, I'd like to point out that only three upstream dialects have a separate `Enums.h`.

```
❯ find . | rg 'Enums.h'
./SPIRV/IR/SPIRVEnums.h
./ArmSME/IR/ArmSMEEnums.h
./SparseTensor/IR/Enums.h
```